1. Last week you wrote a program that printed out a cheery greeting including your
name. Take a copy of it, and modify it so that the user enters their name at the
keyboard, and then receives a greeting.

In [2]:
name = input("Hello, what is your name? ")
print(f"Hello, {name}. Good to meet you!")

Hello, what is your name?  Albatros


Hello, Albatros. Good to meet you!


2. Write a program that prompts a user to enter a temperature in Celsius, and then
displays the corresponding temperature in Fahrenheit, like so:
Enter a temperature in Celsius: 32.5
32.5C is equivalent to 90.5F.

In [3]:
Celsius = float(input("Enter temperature in celsius: "))
Fahrenheit = (Celsius * 9/5) + 32
print(f"{Celsius}C is equivalent to {Fahrenheit}F.")


Enter temperature in celsius:  45.2


45.2C is equivalent to 113.36F.


3. The Head of Computing at the University of Poppleton is tasked with dividing a
group of students into lab groups. A lab group is usually 24 students, but this is
sometimes varied to create groups of similar size. Write a program that prompts for
the number of students and group size, and then displays how many groups will be
needed and how many will be left over in a smaller group.
How many students? 113
Required group size? 22
There will be 5 groups with 3 students left over.
For bonus credit, see if you can fix the grammar in the output. So if there were 101
students in groups of 20 the output would be:
There will be 5 groups with 1 student left over.

In [4]:
students = int(input("How many students? "))
group_size = int(input("Required group size? "))

groups_required = students // group_size
students_left = students % group_size

if students_left ==1:
    print(f"There will be {groups_required} groups with {students_left} student left over.")
else:
    print(f"There will be {groups_required} groups with {students_left} students left over.")

How many students?  31
Required group size?  5


There will be 6 groups with 1 student left over.


4. A kindly teacher wishes to distribute a tub of sweets between her pupils. She will
first count the sweets and then divide them according to how many pupils attend
that day. Write a program that will tell the teacher how many sweets to give to each
pupil, and how many she will have left over.

In [5]:
pupils = int(input("Enter the number of pupils present: "))
sweets_available = int(input("Enter the amount of sweets you have: "))

sweet_to_distribute = sweets_available // pupils
sweets_left = sweets_available % pupils

print(f"You must distribute {sweet_to_distribute} sweets to each pupil, you will have {sweets_left} sweets left over.") 

Enter the number of pupils present:  6
Enter the amount of sweets you have:  30


You must distribute 5 sweets to each pupil, you will have 0 sweets left over.
